src/raw.rs

```diff
@@ -1,15 +1,21 @@
 use alloc::vec::Vec;
 use num_bigint::BigUint;
 use rand_core::CryptoRngCore;
-use zeroize::Zeroize;
+use zeroize::Zeroizing;
 
 use crate::errors::Result;
 use crate::internals;
 use crate::key::{RsaPrivateKey, RsaPublicKey};
 
 pub trait EncryptionPrimitive {
     /// Do NOT use directly! Only for implementors.
-    fn raw_encryption_primitive(&self, plaintext: &[u8], pad_size: usize) -> Result<Vec<u8>>;
+    fn raw_encryption_primitive(&self, plaintext: &[u8], pad_size: usize) -> Result<Vec<u8>> {
+        let int = Zeroizing::new(BigUint::from_bytes_be(plaintext));
+        self.raw_int_encryption_primitive(&int, pad_size)
+    }
+
+    fn raw_int_encryption_primitive(&self, plaintext: &BigUint, pad_size: usize)
+        -> Result<Vec<u8>>;
```

Comment:
I'm not super enthusiastic about this API change, but I did it this way to reduce overall churn (i.e. so I don't have to change all of the existing usages of `raw_encryption_primitive`/`raw_decryption_primitive`.

I'm also not sure what these traits are for... are they intended to be implemented by e.g. hardware accelerators?

Reply:
I think it might be better to drop these traits. Hardware accelerators will most probably implement the `src/key.rs` API rather than `src/raw.rs`.

Reply:
This change looks good to me. Let's land it, I'll post the trait removal on top of it.

Reply:
Cool, can cut a prerelease after that